src/utils/clone.ts

```diff
@@ -0,0 +1,31 @@
+export function cloneShadowRoots(sourceNode: Node, targetNode: Node) {
+    if(sourceNode instanceof Element && targetNode instanceof Element) {
+        const shadowRoot = sourceNode.shadowRoot;
+        if (shadowRoot !== null) {
+            // Create a new Shadow Root for the target element
+            const copiedShadowRoot = targetNode.attachShadow({mode: shadowRoot.mode});
+
+            // Clone the contents of the original ShadowRoot manually
+            const originalShadowContent = shadowRoot.innerHTML;
+
+            // Create a temporary container to hold the content to copy
+            const tempContainer = targetNode.ownerDocument.createElement(shadowRoot.children[0].tagName);
```

Comment:
Will the shadow root always have at least one child?

Reply:
Simplified that... the old code version was a bit hacky. Much cleaner now.

Reply:
`TreeWalker` is nice!